src/components/QueryOutput/QueryOutput.js

```diff
@@ -851,7 +877,87 @@ export class QueryOutput extends React.Component {
 
     return undefined
   }
-
+  getTrainingData = () => {
```

Comment:
I think ideally we will want to move this function outside of the component, and pass in any required data through function params. We will want to duplicate this functionality in vanilla so the more we can separate the better